### PR TITLE
Patch vulnerabilities

### PR DESCRIPTION
**Pull request by choffmeister**
https://avd.aquasec.com/nvd/cve-2022-45061
https://avd.aquasec.com/nvd/cve-2022-41717

### CODE REVIEW

File: integrationtests/volumes_test.go [hunk 1]
```diff
@@ -202,20 +201,22 @@ func TestVolumeResize(t *testing.T) {
...
-			if size, err := getFakeDeviceSizeKilobytes(targetPath); err != nil {
+			finalSize, err := getFakeDeviceSizeKilobytes(targetPath)
+			if err != nil {
 				t.Fatal(err)
-			} else if size != test.finalSize {
-				t.Fatal(fmt.Errorf("expected final size of %d KB, got %d KB", test.finalSize, size))
+			}
+
+			if math.Round(float64(finalSize)/float64(initialSize)) != 2.0 {
+				t.Fatal(fmt.Errorf("expected final size to be roughly double of initial size (final size %d KB, initial size %d KB)", finalSize, initialSize))
```
Comment (choffmeister) [inline integrationtests/volumes_test.go:219]: Took the chance to make test lest relying on exact kB values (see #322)

File: integrationtests/volumes_test.go [hunk 2]
```diff
@@ -96,7 +97,7 @@ func TestVolumePublishUnpublish(t *testing.T) {
 			logger := log.NewLogfmtLogger(NewTestingWriter(t))
 			mountService := volumes.NewLinuxMountService(logger)
 			cryptSetup := volumes.NewCryptSetup(logger)
-			device, err := createFakeDevice("fake-"+test.name, 32)
+			device, err := createFakeDevice("fake-"+test.name, 512)
```
[the remark this answers is not in the document]
Comment (apricote) [reply integrationtests/volumes_test.go:100]: I also stumbled upon this yesterday while working on #353 with xfsprogs `5.19`

**Comment by apricote** [comment]
FYI it looks like our e2e tests broke across all current PRs: https://github.com/hetznercloud/csi-driver/issues/361

**Comment by choffmeister** [comment]
@apricote Ah... Ok. was wondering how the alpine base image upgrade could break the go compiling outside of the image 😄  Probably a change on your self-hosted runners I guess. Then I will wait with the PR until fixed.

**Comment by apricote** [comment]
@choffmeister E2E tests work again! Do you have time to work on this in the coming days? Otherwise I would like to take it over to include this in the next release.

**Comment by apricote** [comment]
I see following error in the logs:

```
2023-01-17T08:10:39.0836088Z Jan 17 08:10:38.861: INFO: At 2023-01-17 08:03:57 +0000 UTC - event for pod-1b72020c-b4af-4e0e-8267-808c56d63b10: {kubelet cluster-worker-1-gh-3937156685-7862460890082335421} FailedMount: MountVolume.SetUp failed for volume "pvc-17d5f88f-0fb4-4d36-ab06-eb2f596304ad" : rpc error: code = Internal desc = failed to publish volume: mount failed: exit status 255
2023-01-17T08:10:39.0836685Z Mounting command: mount
2023-01-17T08:10:39.0837363Z Mounting arguments: -t xfs /dev/disk/by-id/scsi-0HC_Volume_26777133 /var/lib/kubelet/pods/a3c83108-5f37-4b40-9c50-5bb8fc1d0bcb/volumes/kubernetes.io~csi/pvc-17d5f88f-0fb4-4d36-ab06-eb2f596304ad/mount
2023-01-17T08:10:39.0838288Z Output: mount: mounting /dev/disk/by-id/scsi-0HC_Volume_26777133 on /var/lib/kubelet/pods/a3c83108-5f37-4b40-9c50-5bb8fc1d0bcb/volumes/kubernetes.io~csi/pvc-17d5f88f-0fb4-4d36-ab06-eb2f596304ad/mount failed: Invalid argument
```

**Comment by choffmeister** [comment]
@apricote No problem. Happy to create a separate PR for the improve integration test with PR #366

**Pull request closed without merging**